contracts/strategies/morpho/MorphoSupplyVaultStrategy.sol

```diff
@@ -19,50 +24,79 @@ abstract contract MorphoSupplyVaultStrategy is ERC4626Strategy {
     address public poolToken;
 
     /// @notice reward token address (e.g. COMP)
-    /// @dev set to address(0) to skip `redeemRewards`
     address public rewardToken;
 
+    /// @notice MORPHO reward distributor
+    IRewardsDistributor public distributor;
+
     function initialize(
         address _strategyToken,
         address _token,
         address _owner,
         address _poolToken,
-        address _rewardToken
+        address _rewardToken,
+        address _distributor
     ) public {
         AAVE_LENS = IMorphoAaveV2Lens(0x507fA343d0A90786d86C7cd885f5C49263A91FF4);
         COMPOUND_LENS = IMorphoCompoundLens(0x930f1b46e1D081Ec1524efD95752bE3eCe51EF67);
         _initialize(_strategyToken, _token, _owner);
         poolToken = _poolToken;
         rewardToken = _rewardToken;
+        distributor = IRewardsDistributor(_distributor);
     }
 
     /// @notice redeem the rewards
     /// @return rewards amount of reward that is deposited to the ` strategy`
-    function redeemRewards(bytes calldata) public override onlyIdleCDO nonReentrant returns (uint256[] memory rewards) {
-        address _rewardToken = rewardToken;
+    function redeemRewards(bytes calldata data)
+        public
+        virtual
+        override
+        onlyIdleCDO
+        nonReentrant
+        returns (uint256[] memory rewards)
+    {
+        // MORPHO + rewardToken
+        rewards = new uint256[](rewardToken != address(0) ? 2 : 1);
```

Comment:
I think you could always set array length to 1 here if you only manage MORPHO in the default implementation while additional rewards are managed in the child contracts. Alternatively you could port the whole `redeemRewards` of the Morpho Compound integration here which should be better